packages/amplify-provider-awscloudformation/src/pre-push-cfn-processor/cfn-pre-processor.ts

```diff
@@ -24,3 +27,115 @@ export async function preProcessCFNTemplate(filePath: string): Promise<string> {
   await writeCFNTemplate(cfnTemplate, newPath, { templateFormat });
   return newPath;
 }
+
+export async function writeCustomPoliciesToCFNTemplate(
+  resourceDir,
+  cfnFile,
+  category,
+  filePath
+) {
+  const { templateFormat, cfnTemplate } = await readCFNTemplate(path.join(resourceDir, cfnFile));
+  const customPolicies = await stateManager.getCustomPolicies(resourceDir);
+  if (customPolicies === undefined) return;
+
+  if(category === 'function') {
+    await addCustomPoliciesToCFNTemplateForFunction(customPolicies, cfnTemplate, filePath, {templateFormat} );
+  }
+  if(category === 'api') {
```

Comment:
use service to filter out resources for which custom iam policies isn't available. For Lambda functions the service field is 'Lambda' and for API containers is 'ElasticContainers'

Reply:
Done